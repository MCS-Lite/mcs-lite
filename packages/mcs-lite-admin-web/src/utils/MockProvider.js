import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';
import { theme } from 'mcs-lite-theme';

const MockProvider = ({ children }) =>
  <IntlProvider messages={{}} locale="en">
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </IntlProvider>;

MockProvider.propTypes = {
  children: PropTypes.node,
};

export default MockProvider;
