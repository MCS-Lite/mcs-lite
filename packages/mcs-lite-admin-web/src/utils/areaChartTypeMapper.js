// @flow
import R from 'ramda';

type Mapper = string => 'step' | 'linear';

const areaChartTypeMapper: Mapper = R.ifElse(
  R.contains(R.__, ['Switch', 'GPIO']), // eslint-disable-line
  R.always('step'),
  R.always('linear'),
);

export default areaChartTypeMapper;
