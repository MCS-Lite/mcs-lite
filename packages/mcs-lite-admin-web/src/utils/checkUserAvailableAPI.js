/* global fetch */

const checkUserAvailableAPI = ({ accessToken, email }) =>
  fetch(`/api/users/available?email=${email}`, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
  }).then(response => response.json());

export default checkUserAvailableAPI;
