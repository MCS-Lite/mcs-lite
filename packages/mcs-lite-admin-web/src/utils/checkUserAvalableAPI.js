/* global fetch */

import R from 'ramda';

const checkUserAvalableAPI = ({ accessToken, email }) =>
  fetch(`/api/users?email=${email}`, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
  })
    .then(response => response.json())
    .then(R.isEmpty);

export default checkUserAvalableAPI;
