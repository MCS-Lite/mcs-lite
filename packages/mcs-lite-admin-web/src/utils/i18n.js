import { addLocaleData } from 'react-intl';
import en from 'react-intl/locale-data/en';
import zh from 'react-intl/locale-data/zh';

/**
 * Languages supports for react-intl
 * ref: https://github.com/yahoo/react-intl/wiki#locale-data-in-browsers
 */

addLocaleData([...en, ...zh]);
addLocaleData({ locale: 'zh-TW', parentLocale: 'zh' });
addLocaleData({ locale: 'zh-CN', parentLocale: 'zh-TW' });
