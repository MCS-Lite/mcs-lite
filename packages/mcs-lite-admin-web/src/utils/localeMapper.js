import R from 'ramda';

const localeMapper = R.memoize(defaultLocale =>
  R.cond([
    [R.test(/^en/gi), R.always('en')],
    [R.test(/^zh-TW/gi), R.always('zh-TW')],
    [R.test(/^zh-CN/gi), R.always('zh-CN')],
    [R.test(/^zh/gi), R.always('zh-TW')],
    [R.T, R.always(defaultLocale)],
  ]),
);

export default localeMapper;
