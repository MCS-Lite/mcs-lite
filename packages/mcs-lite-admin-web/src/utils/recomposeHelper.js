import { Observable } from 'rxjs/Observable';
import { componentFromStreamWithConfig } from 'recompose/componentFromStream';
import { createEventHandlerWithConfig } from 'recompose/createEventHandler';

export const config = {
  fromESObservable: Observable.from,
  toESObservable: stream => stream,
};

export const componentFromStream = componentFromStreamWithConfig(config);
export const createEventHandler = createEventHandlerWithConfig(config);
