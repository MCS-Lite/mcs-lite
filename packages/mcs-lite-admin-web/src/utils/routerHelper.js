// @flow

type Func = string => Object => Object;

/**
 * For react-router <Link />
 * ref: https://github.com/ReactTraining/react-router/blob/master/docs/API.md#props-1
 *
 * // Function returning location descriptor.
 * <Link to={location => ({ ...location, query: { name: 'ryan' } })}>
 *   Hello
 * </Link>
 *
 * @author Michael Hsu
 */

export const updatePathname: Func = pathname => location => {
  const locale = location.query.locale;
  return { pathname, query: { ...location.query, locale } };
};

export const updateLocale: Func = locale => location => {
  const pathname = location.pathname;
  return { pathname, query: { ...location.query, locale } };
};
