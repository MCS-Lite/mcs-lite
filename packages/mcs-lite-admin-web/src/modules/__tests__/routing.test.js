/* eslint key-spacing: 0 */

import { push } from 'react-router-redux';
import reducer, { constants, actions, cycles } from '../routing';
import { assertSourcesSinks } from '../../utils/helpers';

describe('routing - 1. Constants', () => {
  it('should return constants', () => {
    expect(constants).toMatchSnapshot();
  });
});

describe('routing - 2. Action Creators', () => {
  it('should return actions', () => {
    expect(actions).toMatchSnapshot();
  });
});

describe('ui - 3. Cycle', () => {
  it('should handle Case1: defaultLocale', done => {
    const stateSource = {
      i: {
        routing: {
          locationBeforeTransitions: null,
        },
      },
      s: {
        routing: {
          locationBeforeTransitions: {
            pathname: '/a',
            query: {},
          },
        },
      },
    };

    const actionSink = {
      x: push({ pathname: '/a', query: { locale: 'zh-TW' } }),
    };

    // prettier-ignore
    assertSourcesSinks(
      {
        STATE:  { 'is|': stateSource },
      },
      { ACTION: { '-x|': actionSink } },
      cycles.localeCycle, done,
    );
  });

  it('should handle Case2: Keep last local state with push()', done => {
    const stateSource = {
      i: {
        routing: {
          locationBeforeTransitions: null,
        },
      },
      s: {
        routing: {
          locationBeforeTransitions: {
            pathname: '/a',
            query: {},
          },
        },
      },
      t: {
        routing: {
          locationBeforeTransitions: {
            pathname: '/a',
            query: { locale: 'en' },
          },
        },
      },
      u: {
        routing: {
          locationBeforeTransitions: {
            pathname: '/logout',
            query: {},
          },
        },
      },
    };

    const actionSink = {
      x: push({ pathname: '/a', query: { locale: 'zh-TW' } }),
      y: push({ pathname: '/a', query: { locale: 'en' } }),
      z: push({ pathname: '/logout', query: { locale: 'en' } }),
    };

    // prettier-ignore
    assertSourcesSinks(
      {
        STATE:  { 'istu|': stateSource },
      },
      { ACTION: { '-xyz|': actionSink } },
      cycles.localeCycle, done,
    );
  });
});

describe('routing - 4. Reducer', () => {
  it('should return the initial state', () => {
    const state = reducer();
    expect(state).toMatchSnapshot();
  });

  it('should handle LOCATION_CHANGE', () => {
    const state = reducer(
      {},
      {
        type: constants.LOCATION_CHANGE,
        payload: { pathname: '/' },
      },
    );
    expect(state).toMatchSnapshot();
  });
});
