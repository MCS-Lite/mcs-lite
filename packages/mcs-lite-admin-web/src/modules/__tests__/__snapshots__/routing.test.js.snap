exports[`routing - 1. Constants should return constants 1`] = `
Object {
  "LOCATION_CHANGE": "@@router/LOCATION_CHANGE",
  "PUSH_LOCALE": "mcs-lite-mobile-web/routing/PUSH_LOCALE",
  "PUSH_PATHNAME": "mcs-lite-mobile-web/routing/PUSH_PATHNAME",
}
`;

exports[`routing - 2. Action Creators should return pushLocale actions 1`] = `
Object {
  "payload": "en",
  "type": "mcs-lite-mobile-web/routing/PUSH_LOCALE",
}
`;

exports[`routing - 2. Action Creators should return pushPathname actions 1`] = `
Object {
  "payload": "/login",
  "type": "mcs-lite-mobile-web/routing/PUSH_PATHNAME",
}
`;

exports[`routing - 4. Reducer should handle LOCATION_CHANGE 1`] = `
Object {
  "locationBeforeTransitions": Object {
    "pathname": "/",
  },
}
`;

exports[`routing - 4. Reducer should return the initial state 1`] = `
Object {
  "locationBeforeTransitions": null,
}
`;
